### PR TITLE
Add support for BranchListOptions

### PR DESCRIPTION
**Pull request by shotat**
Add support for BranchListOptions for `RepositoriesService.ListBranches`.

ref. https://developer.github.com/v3/repos/branches/#list-branches

### CODE REVIEW

**Comment by shotat** [comment]
This branch failed to pass CI because of Go 1.13.
Could anyone fix the CI settings?

File: github/repos.go [hunk 1]
```diff
@@ -125,6 +125,18 @@ func (r Repository) String() string {
 	return Stringify(r)
 }
 
+// BranchListOptions specifies the optional parameters to the
+// RepositoriesService.ListBranches method.
+type BranchListOptions struct {
+	// Setting to true returns only protected branches.
+	// When set to false, only unprotected branches are returned.
+	// Omitting this parameter returns all branches.
+	// Default: null
```
Comment (gmlewis) [inline github/repos.go]: In Go, we prefer to say `nil` over `null`. Could you please change this?

**Comment by shotat** [comment]
Thanks for your review!

**Comment by gmlewis** [comment]
Thank you, @gauntface!
Merging.

**Pull request merged**